LSL/liblsl/CMakeLists.txt

```diff
@@ -1,36 +1,165 @@
+	src/sample.cpp
+	src/sample.h
+	src/send_buffer.cpp
+	src/send_buffer.h
+	src/socket_utils.cpp
+	src/socket_utils.h
+	src/stream_info_impl.cpp
+	src/stream_info_impl.h
+	src/stream_inlet_impl.h
+	src/stream_outlet_impl.cpp
+	src/stream_outlet_impl.h
+	src/tcp_server.cpp
+	src/tcp_server.h
+	src/time_postprocessor.cpp
+	src/time_postprocessor.h
+	src/time_receiver.cpp
+	src/time_receiver.h
+	src/udp_server.cpp
+	src/udp_server.h
+)
+set(headers
+	include/lsl_c.h
+	include/lsl_cpp.h
+)
+
+set(target lsl)
+if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+	set(lslplatform 64)
+else()
+	set(lslplatform 32)
+endif()
+set(lslplatform ${lslplatform}_${CMAKE_CXX_COMPILER_ID}${MSVC_VERSION})
 
-# Build shared and/or static library
-option (BUILD_SHARED "Build shared library." ON)
-option (BUILD_STATIC "Build static library." OFF)
+set(lsl_export_targets ${target})
 
-if (USE_SYSTEM_BOOST)
-  find_package(Boost COMPONENTS chrono filesystem serialization system thread REQUIRED)
+if (LSL_USE_SYSTEM_BOOST)
+	find_package(Boost COMPONENTS chrono filesystem serialization system thread REQUIRED)
+	set(LSL_BOOST_LIBS Boost::filesystem Boost::serialization Boost::thread)
 else ()
-  add_subdirectory (external)
-endif (USE_SYSTEM_BOOST)
+	add_subdirectory (external)
+	set(LSL_BOOST_LIBS "lslboost")
+endif ()
+
+# common definitions for the lsllib variants
+function(lsllib_properties libname)
+	target_include_directories(${libname}
+		PRIVATE
+		${CMAKE_CURRENT_BINARY_DIR} # generated export header
+		INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+	)
+    message(STATUS "Linking LSL against: ${LSL_BOOST_LIBS}")
+	message(STATUS "Found Boost libraries: ${Boost_LIBRARIES}")
+	target_link_libraries(${libname} PRIVATE ${LSL_BOOST_LIBS})
+endfunction()
+
+
+# shared library
+add_library(${target} SHARED ${sources} ${headers})
+lsllib_properties(${target})
+set_target_properties(${target} PROPERTIES OUTPUT_NAME "${target}${lslplatform}")
+
+if(${LSL_BUILD_STATIC})
+	add_library(${target}-static ${sources} ${headers})
+	lsllib_properties(${target}-static)
+	list(APPEND lsl_export_targets "${target}-static")
+	# for LSL_CPP_API export header
+	target_compile_definitions(${target}-static PRIVATE LSL_STATIC_DEFINE)
+	set_target_properties(${target}-static PROPERTIES
+		OUTPUT_NAME "${target}${lslplatform}-static")
+endif()
+
+# Generate with LIBLSL_CPP_API exports
+include(GenerateExportHeader)
+generate_export_header(${target}
+	BASE_NAME LSL
+	EXPORT_MACRO_NAME LIBLSL_CPP_API
+	EXPORT_FILE_NAME lsl_export_defines.h)
+
+set_target_properties(${target} PROPERTIES
+	VERSION ${LSL_VERSION_MAJOR}.${LSL_VERSION_MINOR}.${LSL_VERSION_PATCH}
+	SOVERSION ${LSL_VERSION_MAJOR}.0.0 # hopefully backwards compatible symlink
+	CXX_VISIBILITY_PRESET hidden # only export symbols marked with LIBLSL_CPP_API
+)
+
+# TODO: Need to check if all flags are necessary or if some flags are missing
+if (WIN32)
+	if (MINGW)
+		target_link_libraries (${target} winmm ws2_32 wsock32)
```

Comment:
I think cross-compilers (from Linux to Windows) will use MINGW, so it should probably be kept.